Issue#226 get polls

AriannaLiss opened the pull request:
dev
## ZenHub

* [Main ZenHub ticket](https://app.zenhub.com/workspaces/home-project-5f7b77ff8db73a001cb17009)


## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

Miss an ability to get (get, get all) polls independently from cooperation ([#226](https://app.zenhub.com/workspaces/home-project-5f7b77ff8db73a001cb17009/issues/ita-social-projects/home/226))

## Summary of change

Added get methods for /polls and /polls/{id} endpoints.

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

AriannaLiss committed:
Issue#226: added /polls and /polls/{id} endpoints.

Added parameter cooperation_id.

AriannaLiss committed:
Issue#226: implemented PollsApi.

Created PollsApiImpl.
Implemented methods getPoll and queryPoll.

AriannaLiss committed:
Issue#226: added getPollTest and getNonExistingPollTest in CooperationPollApiIT.

AriannaLiss committed:
Issue#226: added query tests for /polls.

Added IQueryPoll.
Reorganized polls/QueryCooperationPollIT.
Added polls/QueryPollIT.
Added query/PollQuery.

kh0ma commented on home-open-api/src/main/resources/yaml/openapi.yaml:
```diff
@@ -1171,6 +1173,57 @@ paths:
+        - $ref: '#/components/parameters/page_size'
+        - $ref: '#/components/parameters/sort'
+        - $ref: '#/components/parameters/filter'
+        - $ref: '#/components/parameters/cooperation_id'
```
Is that parameter mandatory here?

kh0ma replied:
Let me put the idea of why I'm asking it here. All the resources should be cooperation scoped. So we need to be sure that we are following it.

AriannaLiss replied:
Ok, I'll make it mandatory.

AriannaLiss committed:
Issue#226: made parameter cooperation_id required in openapi (for polls).

Moved cooperation_id on the first position in a list of parameters in openapi /polls, in PollsApiImpl queryPoll() and in PollQuery perform().

AriannaLiss committed:
Issue#226: refactor IQueryPoll

Moved realization getAllPollsFromNotExistingCooperation to QueryCooperationPollIT and QueryPollIT.

Added cooperationId in queries in QueryPollIT.

Made verification in QueryPollIT.getAllPollsOnlyByPollId if using queryPoll without mandatory cooperationId throws exception.

AriannaLiss committed:
Issue#226: disabled two tests and created tasks 250 and 251.

Disabled QueryPollIT.getAllPollsOnlyByPollId and created task 250.
Disabled QueryCooperationPollIT.getAllPollsFromNotExistingCooperation and created task 251.

AriannaLiss committed:
Issue#226: renamed IQueryPoll to QueryPoll and change it to abstract class from interface.

AriannaLiss committed:
Merge remote-tracking branch 'origin/dev' into Issue#226-Get-polls

# Conflicts:
#	home-open-api/src/main/resources/yaml/openapi.yaml

AriannaLiss committed:
Issue#226: Fixed after merge conflict.

Returned description cooperation_id in openapi.
PollsApiImpl moved to PollApiImpl.
